department-of-veterans-affairs/caseflow
# LiannaAndJeremy/APPEALS-4293: Pre-Population of Form Fields of Queue's Hearing Conversion Form

Resolves [APPEALS-4293](https://vajira.max.gov/browse/APPEALS-4293)

### Description
As a VSO User, I need the applicable fields on the convert-to-virtual form to prepopulate with the data from their data source, so the info can be reviewed and/or updated before opting the appellant into a virtual hearing.

### Acceptance Criteria
- [x] The following fields are prepopulated with information from the Corporate Data base  
   - Veteran/Appellant Email 
   - Veteran/Appellant Time Zone 
   - POA/Representative Time Zone
- [x] The values entered into the hearing request type conversion form should be reflected in the payload sent to the backend for persistence.


### Testing Plan
1) Either reseed your database, or ensure that the VSO user you will be using for the test has an email:

```ruby
User.where(css_id: "BILLIE_VSO").update(email: "BillieVSO@test.com")
```

2) Intake a new appeal with a docket type of hearing. Select a hearing type other than virtual (or none at all).

3) For the prompt, "Is the claimant someone other than the Veteran?", select 'Yes', then 'Bob Vance'.

4) Add any issue you'd like to the appeal (other than a VHA or a pre-docket education one).

5) Take note of the appeal's UUID after successfully establishing it through intake. You can get this by clicking "Correct the Issues" on the intake completion page and then grabbing it from the URL, or you can use the Rails console:

```ruby
Appeal.order("created_at").last.uuid
```
6) Switch to a VSO user (ex: `BILLIE_VSO`)

7) Navigate to the Case Details page: `/queue/appeals/[uuid]`

8) Scroll down to the Hearings section, and click on 'Convert to virtual'. Take note of the task id in the url path:

`/queue/appeals/<APPEAL_ID>/tasks/<TASK_ID>/change_hearing_request_type_to_virtual`

9) Ensure that the Appellant/Veteran Email, Appellant/Veteran Timezone, POA Type, POA Name, POA Email, and POA Timezone sections have been pre-populated with the values pertaining to the respective parties.

10) The POA/Representative fields should contain the information of the **current** user.

<img width="363" alt="rep-section" src="https://user-images.githubusercontent.com/99351305/173910626-177a38b5-512b-4f2d-9b1a-aa8ec65bcfb8.PNG">

11) Open up the developer tools for your browser. If you are using Google Chrome, you can use the keyboard shortcut `CTRL+SHIFT+I`. Open the network tab.

12) Change the values of all of the non-disabled fields to ones other than what was pre-populated.

13) Submit the form.

14) Go back to your dev tools window, and select the network transaction with the title being the task ID from step 6. Click on `Payload`, and collapse the various dropdowns until you find the `business_payloads` section. The `business_payloads.email_recipients` dictionary should contain the new values you have just entered.

<img width="544" alt="business_payload" src="https://user-images.githubusercontent.com/99351305/173629389-95be4b6a-ba11-4757-9173-70d47824c8a8.PNG">



### User Facing Changes

 
![vso-queue-form-prepopulation](https://user-images.githubusercontent.com/99351305/173629793-5305b1a5-f0a7-4910-9047-7cf3c6133d9a.gif)
